src/garage/tf/algos/te_npo.py

```diff
@@ -0,0 +1,1191 @@
+            traj_gammas = tf.constant(float(self._discount),
+                                      dtype=tf.float32,
+                                      shape=[self.max_path_length])
+            # pylint false alarm
+            # pylint: disable=no-value-for-parameter
+            traj_discounts = tf.compat.v1.cumprod(traj_gammas,
+                                                  exclusive=True,
+                                                  name='traj_discounts')
+            discount_traj_ll = traj_discounts * traj_ll
+            discount_traj_ll = tf.boolean_mask(discount_traj_ll, i.valid_var)
+
+            with tf.name_scope('loss'):
+                infer_loss = -tf.reduce_mean(discount_traj_ll,
+                                             name='infer_loss')
+
+            with tf.name_scope('kl'):
+                # Calculate predicted encoder distributions for each timestep
+
+                # Calculate KL divergence
+                kl = old_dist.kl_divergence(dist)
+                infer_kl = tf.reduce_mean(kl, name='infer_kl')
+
+            return infer_loss, infer_kl
+
+    def _policy_opt_input_values(self, samples_data):
+        """Map rollout samples to the policy optimizer inputs.
+
+        Args:
+            samples_data (dict): Processed sample data.
+                See process_samples() for details.
+
+        Returns:
+            list(np.ndarray): Flatten policy optimization input values.
+
+        """
+        policy_state_info_list = [
+            samples_data['agent_infos'][k] for k in self.policy.state_info_keys
+        ]
+        embed_state_info_list = [
+            samples_data['latent_infos'][k]
+            for k in self.policy.encoder.state_info_keys
+        ]
+        policy_opt_input_values = self._policy_opt_inputs._replace(
+            obs_var=samples_data['observations'],
+            action_var=samples_data['actions'],
+            reward_var=samples_data['rewards'],
+            baseline_var=samples_data['baselines'],
+            trajectory_var=samples_data['trajectories'],
+            task_var=samples_data['tasks'],
+            latent_var=samples_data['latents'],
+            valid_var=samples_data['valids'],
+            policy_state_info_vars_list=policy_state_info_list,
+            embed_state_info_vars_list=embed_state_info_list,
+        )
+
+        return flatten_inputs(policy_opt_input_values)
+
+    def _inference_opt_input_values(self, samples_data):
+        """Map rollout samples to the inference optimizer inputs.
+
+        Args:
+            samples_data (dict): Processed sample data.
+                See process_samples() for details.
+
+        Returns:
+            list(np.ndarray): Flatten inference optimization input values.
+
+        """
+        infer_state_info_list = [
+            samples_data['trajectory_infos'][k]
+            for k in self._inference.state_info_keys
+        ]
+        # pylint: disable=unexpected-keyword-arg
```

Comment:
can you explain this?

Reply:
Pylint passes on my local machine, but it failed on CI. I removed this line and see how it goes.